packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useRef, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { FormattedMessage } from 'react-intl';
+import { enableFocusStyles, disableFocusStyles, restoreFocusStyles } from './_SwitchUtils';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+const SWITCH_STATE = Object.freeze({
+  ON: <FormattedMessage id="Terra.switch.switchstatuslabel.on" />,
+  OFF: <FormattedMessage id="Terra.switch.switchstatuslabel.off" />,
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button.
+   */
+  switchId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  checked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * The label text of the Switch component.
+   */
+  labelText: PropTypes.string.isRequired,
+  /**
+   * Id of the Switch label.
+   */
+  labelId: PropTypes.string.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Returns Parameters: 1. switch value 2. event.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  checked: false,
+  disabled: false,
+  onChange: undefined,
+};
+
+const Switch = (props) => {
+  const {
+    switchId,
+    labelId,
+    checked,
+    disabled,
+    onChange,
+    labelText,
+    ...customProps
+  } = props;
+
+  const sliderButton = useRef();
+
+  const handleOnClick = useCallback(
+    (event) => {
+      // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
+      // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
+      // This will set focus on the button when clicked.
+      sliderButton.current.focus();
+      if (onChange) {
+        onChange(!checked, event);
```

Comment:
Do we need to pass back `!checked` if the consumers have to manage this state themselves?

Reply:
Yes that was my thought initially. later changed method to return updated state instead of same state w.r.t this [comment](https://github.com/cerner/terra-core/pull/2846#discussion_r383959668)